scenario/state.py

```diff
@@ -895,17 +896,85 @@ def __repr__(self):
             return f"{status_type_name}()"
         return f"{status_type_name}('{self.message}')"
 
+    @classmethod
+    def from_status_name(
+        cls,
+        name: _RawStatusLiteral,
+        message: str = "",
+    ) -> "_EntityStatus":
+        # Note that this won't work for UnknownStatus.
+        # All subclasses have a default 'name' attribute, but the type checker can't tell that.
+        return cls._entity_statuses[name](message=message)  # type:ignore
+
+    @classmethod
+    def from_ops(cls, obj: StatusBase) -> "_EntityStatus":
+        return cls.from_status_name(cast(_RawStatusLiteral, obj.name), obj.message)
+
+
+@dataclasses.dataclass(frozen=True, eq=False, repr=False)
+class UnknownStatus(_EntityStatus, ops.UnknownStatus):
+    __doc__ = ops.UnknownStatus.__doc__
+
+    name: Literal["unknown"] = "unknown"
+
+    def __init__(self):
+        super().__init__(name=self.name)
+        self.__class__._entity_statuses[self.name] = self.__class__
```

Comment:
It probably won't *hurt*, but it seems odd to be registering the name in this class dictionary on every instance init. Could we just populate the class-level dict at the end? Something like:

```python
class ActiveStatus: ...  # and define other status classes

_EntityStatus._classes_by_name = {
    'active': ActiveStatus,
    'error': ErrorStatus,
    ...
}
```

Reply:
Yeah, I think I was putting in too much effort to avoid that. Simple is probably best here.